src/Aspire.Hosting.SurrealDb/SurrealDbBuilderExtensions.cs

```diff
@@ -0,0 +1,84 @@
+    /// <param name="userName">The parameter used to provide the administrator username for the SurrealDB resource.</param>
+    /// <param name="password">The parameter used to provide the administrator password for the SurrealDB resource. If <see langword="null"/> a random password will be generated.</param>
+    /// <param name="port">The host port for the SurrealDB instance.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<SurrealDbServerResource> AddSurrealServer(
+        this IDistributedApplicationBuilder builder,
+        string name,
+        IResourceBuilder<ParameterResource>? userName = null,
+        IResourceBuilder<ParameterResource>? password = null,
+        int? port = null
+    )
+    {
+        // The password must be at least 8 characters long and contain characters from three of the following four sets: Uppercase letters, Lowercase letters, Base 10 digits, and Symbols
+        var passwordParameter = password?.Resource ?? ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, $"{name}-password", minLower: 1, minUpper: 1, minNumeric: 1);
+
+        var surrealServer = new SurrealDbServerResource(name, userName?.Resource, passwordParameter);
+        return builder.AddResource(surrealServer)
+                      .WithEndpoint(port: port, targetPort: 8000, name: SurrealDbServerResource.PrimaryEndpointName)
+                      .WithImage(SurrealDbContainerImageTags.Image, SurrealDbContainerImageTags.Tag)
+                      .WithEnvironment(context =>
+                      {
+                          context.EnvironmentVariables[UserEnvVarName] = surrealServer.UserNameReference;
+                          context.EnvironmentVariables[PasswordEnvVarName] = surrealServer.PasswordParameter;
+                      })
+                      .WithEntrypoint("/surreal")
+                      .WithArgs("start", "--auth");
+    }
+
+    /// <summary>
+    /// Adds a SurrealDB database to the application model. This is a child resource of a <see cref="SurrealDbServerResource"/>.
+    /// </summary>
+    /// <param name="builder">The SurrealDB resource builders.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="namespaceName">The name of the namespace. If not provided, this defaults to "test".</param>
+    /// <param name="databaseName">The name of the database. If not provided, this defaults to "test".</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<SurrealDbDatabaseResource> AddDatabase(this IResourceBuilder<SurrealDbServerResource> builder, string name, string namespaceName = "test", string databaseName = "test")
```

Comment:
We use the `name` parameter as the database name if databaseName is not provided.
We should do the same as Postgres.

https://github.com/dotnet/aspire/blob/7a5ee578710258d3d43ff31c290c33380ee01451/src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs#L125-L134

Reply:
Yes, I saw that. It is indeed easier this way but I am a little stuck since surrealdb offers a namespace feature where a single db can host N namespaces and each namespace can host M databases. Which mean that a named database could be created in different namespaces (it rarely happens but it can happen).

I also had the idea of generating a random name for the namespace but at the same time, that feels weird.

Should I create a `SurrealDbNamespaceResource` to counter that problem? I also had this idea but that bring more code to the table... See: `AddSurrealServer(...).AddNamespace(...).AddDatabase(...)`

Reply:
Ok, I have created a `SurrealDbNamespaceResource` as it feels natural because you can register N namespaces per server and M databases per namespace.